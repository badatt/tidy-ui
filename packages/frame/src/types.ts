import { HTMLAttributes } from 'react';
import { IBaseProps } from '@tidy-ui/types';

/**
 * The IFrameProps interface offers a versatile framework for creating visually appealing frames that encapsulate your content. Whether you're aiming for sharp edges, circular elegance, or streamlined wrapping, this interface provides the tools you need to achieve your design objectives.
 *
 * @interface IFrameProps
 * @type {IFrameProps}
 * @augments {HTMLAttributes<HTMLDivElement>}
 * @public
 * @since 0.47.0
 */
export interface IFrameProps extends IBaseProps, HTMLAttributes<HTMLDivElement> {
  /**
   * Indicate that the content is loaded and ready to be rendered, allowing the frame to gracefully transition or disappear.
   *
   * @type {boolean}
   * @default false
   * @public
   * @since 0.47.0
   */
  isLoaded?: boolean;

  /**
   * Craft frames with sharp edges, adding a touch of modern aesthetics to your design.
   *
   * @type {boolean}
   * @default false
   * @public
   * @since 0.47.0
   */
  isSharp?: boolean;

  /**
   * Opt to wrap child items neatly within the dimensions of the frame, ensuring a visually polished presentation.
   *
   * @type {boolean}
   * @default false
   * @public
   * @since 0.47.0
   */
  isWrapped?: boolean;

  /**
   * Define the size (radius) of circular variants, enabling you to tailor the frame's appearance to your desired specifications.
   *
   * @type {string}
   * @public
   * @since 0.47.0
   */
  size?: string;

  /**
   * Select the frame variant that suits your design vision. Choose between a circular variant ('circle') or a sleek line variant ('line').
   * If not supplied anything, it creates a rectangle box frame with given dimensions.
   *
   * @type {'circle' | 'line'}
   * @public
   * @since 0.47.0
   */
  variant?: 'circle' | 'line';
}
