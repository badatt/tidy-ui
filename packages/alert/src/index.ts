export * from './Alert';
export * from './AlertFooter';
export * from './AlertTitle';
