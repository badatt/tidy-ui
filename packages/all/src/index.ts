export * from '@tidy-ui/commons';
export * from '@tidy-ui/feedback';
export * from '@tidy-ui/form';
export * from '@tidy-ui/layout';
export * from '@tidy-ui/navigation';
export * from '@tidy-ui/presentation';
