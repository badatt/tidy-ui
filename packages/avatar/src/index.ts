export * from './Avatar';
export * from './AvatarBadge';
export * from './AvatarGroup';
