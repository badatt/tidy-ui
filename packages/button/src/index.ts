export * from './Button';
export * from './Anchor';
