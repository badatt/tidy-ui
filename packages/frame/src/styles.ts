import { css } from '@tidy-ui/commons';
import { IFrameProps } from './types';

const circleStyle = css<IFrameProps>`
  border-radius: 50%;
  ${({ size }) => css`
    height: ${size};
    width: ${size};
  `}
`;

const textStyle = css<IFrameProps>`
  height: 1em;
`;

const rectangleStyle = css<IFrameProps>`
  ${({ theme: { layout }, isSharp }) => css`
    ${!isSharp &&
    css`
      border-radius: ${layout.radius};
    `}
  `}
`;

export { circleStyle, rectangleStyle, textStyle };
