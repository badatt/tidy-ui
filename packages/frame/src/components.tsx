import { keyframes } from 'styled-components';
import { applyStandardOverrideStyles, color, css, hsla, ITheme, styled } from '@tidy-ui/commons';
import { circleStyle, rectangleStyle, textStyle } from './styles';
import { IFrameProps } from './types';

/**
 * Creates a blink effect
 *
 * @internal
 */
const frameAnimation = (props: ITheme) => {
  return keyframes`
    from {
      background-color: ${props.theme.isDark ? hsla(color.gray[700]) : hsla(color.gray[300])};
    }
    to {
      background-color: ${props.theme.isDark ? hsla(color.gray[700], 0.2) : hsla(color.gray[200], 0.6)};
    }
  `;
};

const FrameRoot = styled.div<IFrameProps>`
  display: flex;
  height: 100%;
  width: 100%;
  ${({ isLoaded, variant }) => css`
    ${!isLoaded &&
    css`
      animation: 0.8s linear 0s infinite alternate none running ${frameAnimation};
    `}
    ${variant === 'circle' && circleStyle}
    ${variant === 'line' && textStyle}
    ${!variant && rectangleStyle}
  `}
  ${applyStandardOverrideStyles}
`;

export { FrameRoot };
