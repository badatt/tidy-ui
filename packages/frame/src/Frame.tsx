import React from 'react';
import { FrameRoot } from './components';
import { IFrameProps } from './types';

const Frame = React.forwardRef<HTMLDivElement, IFrameProps>((props, ref) => {
  const { children, ...rest } = props;
  const { isLoaded, isWrapped } = rest;

  return (
    <>
      {isLoaded ? (
        <>
          {isWrapped ? (
            <FrameRoot ref={ref} role="frame" {...rest}>
              {children}
            </FrameRoot>
          ) : (
            children
          )}
        </>
      ) : (
        <>
          <FrameRoot ref={ref} role="frame" {...rest}></FrameRoot>
        </>
      )}
    </>
  );
});

Frame.defaultProps = {};

Frame.displayName = 'Frame';

export { Frame };
