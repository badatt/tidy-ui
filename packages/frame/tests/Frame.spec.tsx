/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { orchidDark, orchidLight, TidyUiProvider } from '../../commons/src';
import { Frame } from '../src';

describe('Frame', () => {
  it('Basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Frame />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Dark mode basic render', () => {
    const tree = render(
      <TidyUiProvider theme={orchidDark}>
        <Frame />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Sharp', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Frame isSharp />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Variants', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Frame height="240px" width="350px" />
        <Frame variant="circle" size="240px" />
        <Frame variant="line" />
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Wrapped', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Frame isWrapped>wrapped</Frame>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Loaded', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Frame isLoaded>loaded</Frame>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Loaded & wrapped', () => {
    const tree = render(
      <TidyUiProvider theme={orchidLight}>
        <Frame isLoaded isWrapped>
          loaded and wrapped
        </Frame>
      </TidyUiProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
