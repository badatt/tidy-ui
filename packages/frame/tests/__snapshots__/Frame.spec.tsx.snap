// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame Basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
}

<body>
    <div>
      <div
        class="c0"
        role="frame"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
}

<div>
    <div
      class="c0"
      role="frame"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Frame Dark mode basic render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
}

<body>
    <div>
      <div
        class="c0"
        role="frame"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
}

<div>
    <div
      class="c0"
      role="frame"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Frame Loaded & wrapped 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  border-radius: 0.5rem;
}

<body>
    <div>
      <div
        class="c0"
        role="frame"
      >
        loaded and wrapped
      </div>
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  border-radius: 0.5rem;
}

<div>
    <div
      class="c0"
      role="frame"
    >
      loaded and wrapped
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Frame Loaded 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      loaded
    </div>
  </body>,
  "container": <div>
    loaded
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Frame Sharp 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
}

<body>
    <div>
      <div
        class="c0"
        role="frame"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
}

<div>
    <div
      class="c0"
      role="frame"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Frame Variants 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
  height: 240px;
  width: 350px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 50%;
  height: 240px;
  width: 240px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  height: 1em;
}

<body>
    <div>
      <div
        class="c0"
        height="240px"
        role="frame"
        width="350px"
      />
      <div
        class="c1"
        role="frame"
      />
      <div
        class="c2"
        role="frame"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
  height: 240px;
  width: 350px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 50%;
  height: 240px;
  width: 240px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  height: 1em;
}

<div>
    <div
      class="c0"
      height="240px"
      role="frame"
      width="350px"
    />
    <div
      class="c1"
      role="frame"
    />
    <div
      class="c2"
      role="frame"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Frame Wrapped 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
}

<body>
    <div>
      <div
        class="c0"
        role="frame"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  -webkit-animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  animation: 0.8s linear 0s infinite alternate none running dhGBQd;
  border-radius: 0.5rem;
}

<div>
    <div
      class="c0"
      role="frame"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
